src/main/cpp/blaze.cc

```diff
@@ -893,9 +893,11 @@ class ExtractBlazeZipProcessor : public PureZipExtractorProcessor {
   void Process(const char *filename, const devtools_ijar::u4 attr,
                const devtools_ijar::u1 *data, const size_t size) override {
     string path = blaze_util::JoinPath(embedded_binaries_, filename);
-    if (!blaze_util::MakeDirectories(blaze_util::Dirname(path), 0777)) {
-      BAZEL_DIE(blaze_exit_code::INTERNAL_ERROR)
-          << "couldn't create '" << path << "': " << GetLastErrorString();
+    if (created_directories_.insert(path).second) {
```

Comment:
It's not obvious why this is here, I would add a comment here explaining that you do this to prevent duplicate work for performance reasons. Otherwise, I could imagine someone accidentally removing this without testing the performance implications.

Reply:
Done.